### PR TITLE
Make ROTC year optional

### PR DESCRIPTION
**Pull request by alexose**
Tiny lil' fix for #3351.

### CODE REVIEW

**Review by akainic — COMMENTED** [review]
Looks good, but also will need to update the validation function for this page to not require the commission year.

**Comment by alexose** [comment]
Whoops!  Fixed.

File: src/js/edu-benefits/utils/validations.js [hunk 1]
```diff
@@ -322,8 +322,7 @@ function isValidRotcScholarshipAmount(data) {
 }
 
 function isValidRotcHistoryPage(data) {
-  return data.seniorRotcCommissioned.value !== 'Y' || (isNotBlank(data.seniorRotc.commissionYear.value)
-    && data.seniorRotc.rotcScholarshipAmounts.every(isValidRotcScholarshipAmount));
+  return data.seniorRotcCommissioned.value !== 'Y' && data.seniorRotc.rotcScholarshipAmounts.every(isValidRotcScholarshipAmount);
```
Comment (akainic) [inline src/js/edu-benefits/utils/validations.js]: This should be `||` not `&&`. Either they weren't commissioned through ROTC or they were in which case the scholarship amounts need to be valid.

**Review by akainic — COMMENTED** [review]
Little fix for validation logic

File: src/js/edu-benefits/utils/validations.js (hunk 1 above)
Comment (alexose) [inline src/js/edu-benefits/utils/validations.js]: Erp, right.  The `!==` threw me.   Fixed

File: src/js/edu-benefits/components/military-history/RotcHistoryFields.jsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ export default class RotcHistoryFields extends React.Component {
             onValueChange={(update) => {this.props.onStateChange('seniorRotcCommissioned', update);}}/>
         <div>
           <div className="input-section">
-            <ErrorableNumberInput required
+            <ErrorableNumberInput
                 additionalClass="usa-input-medium"
                 errorMessage={validateIfDirty(this.props.data.seniorRotc.commissionYear, isValidYear) ? undefined : 'Please enter a valid year'}
```
Comment (jbalboni) [inline src/js/edu-benefits/components/military-history/RotcHistoryFields.jsx]: Do you get an invalid year message if you focus and then blur the field without entering anything? I wonder if this needs to be changed to skip isValidYear if the year is blank.

**Review by jbalboni — APPROVED** [review]
One question, but LGTM.

File: src/js/edu-benefits/components/military-history/RotcHistoryFields.jsx (hunk 2 above)
Comment (alexose) [inline src/js/edu-benefits/components/military-history/RotcHistoryFields.jsx]: Good find-- I added a new validation for this case.

**Comment by jbalboni** [comment]
👍

**Pull request merged**